src/Serilog/Formatting/Json/JsonValueFormatter.cs

```diff
@@ -172,7 +172,11 @@ protected virtual void FormatLiteralValue(object? value, TextWriter output)
 
         if (value is ValueType)
         {
-            if (value is int or uint or long or ulong or decimal or byte or sbyte or short or ushort)
+            if (value is int or uint or long or ulong or decimal or byte or sbyte or short or ushort
+#if FEATURE_HALF
+                or Half
```

Comment:
Since `Half` can be infinite or NaN (AFAIK), we should probably handle it here in the same way as `double` and `float` are handled below